File: Examples/src/programmersguide/workingwithtasks/wbsassociatedwithtask/java/WbsAssociatedWithTask.java
/* 
 * Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Slides. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package programmersguide.workingwithtasks.wbsassociatedwithtask.java;

import com.aspose.tasks.*;

public class WbsAssociatedWithTask
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
	String dataDir = "src/programmersguide/workingwithtasks/wbsassociatedwithtask/data/";  

	Project project = new Project(dataDir + "input.mpp");

	// Create a ChildTasksCollector instance
	ChildTasksCollector collector = new ChildTasksCollector();

	// Collect all the tasks from RootTask using TaskUtils
	TaskUtils.apply(project.getRootTask(), collector, 0);

	//Parse through all the collected tasks
	for (Task tsk:collector.getTasks())
	{
	    System.out.println(tsk.get(Tsk.WBS));
	    System.out.println(tsk.get(Tsk.WBS_LEVEL));
	    // set custom WBS
	    tsk.set(Tsk.WBS, "custom wbs");
	}  
    }
}

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/WbsAssociatedWithTask.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\ChildTasksCollector as ChildTasksCollector;
use com\aspose\tasks\TaskUtils as TaskUtils;
use com\aspose\tasks\Tsk as Tsk;
class WbsAssociatedWithTask{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

        # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);

		$tsk = new Tsk();

		$tasks = $collector->getTasks();

		# Parse through all the collected tasks
		$i = 0;
		while ($i < sizeof($tasks)) {
            $task = $tasks -> get($i);
			print "WBS: " . (string)$task -> get($tsk -> WBS).PHP_EOL;// . to_string
			print "WBS Level: " . (string)$task -> get($tsk -> WBS_LEVEL).PHP_EOL;// . to_s
			$i += 1;
		}

    }

}